spring-pulsar/src/main/java/org/springframework/pulsar/listener/PulsarContainerProperties.java

```diff
@@ -246,4 +257,35 @@ public void setPulsarConsumerProperties(Properties pulsarConsumerProperties) {
 		this.pulsarConsumerProperties = pulsarConsumerProperties;
 	}
 
+	@SuppressWarnings("unchecked")
+	public void updateContainerProperties() {
+		if (!this.pulsarConsumerProperties.isEmpty()) {
+			if (this.pulsarConsumerProperties.containsKey(SUBSCRIPTION_NAME)) {
+				this.subscriptionName = (String) this.pulsarConsumerProperties.get(SUBSCRIPTION_NAME);
+			}
+			if (this.pulsarConsumerProperties.containsKey(SUBSCRIPTION_TYPE)) {
+				this.subscriptionType = (SubscriptionType) this.pulsarConsumerProperties.get(SUBSCRIPTION_TYPE);
+			}
+			if (this.pulsarConsumerProperties.containsKey(TOPIC_NAMES)) {
+				Object topicNamesFromConsumerProperties = this.pulsarConsumerProperties.get(TOPIC_NAMES);
```

Comment:
These are set in code via
```java
		JavaUtils.INSTANCE.acceptIfNotNull(this.phase, instance::setPhase)
				.acceptIfNotNull(this.applicationContext, instance::setApplicationContext)
				.acceptIfNotNull(this.applicationEventPublisher, instance::setApplicationEventPublisher)
				.acceptIfNotNull(endpoint.getConsumerProperties(),
						instance.getContainerProperties()::setPulsarConsumerProperties);
```
from `org.springframework.pulsar.config.AbstractPulsarListenerContainerFactory#initializeContainer`. A step through debugger would tell what it actually is coming in as. This may only be coming in differently due to a test. If that is the case we could adjust the test instead.